llama_stack/cli/stack/build.py

```diff
@@ -203,28 +268,16 @@ def _run_stack_build_command_from_build_config(
         build_file_path = build_dir / f"{build_config.name}-build.yaml"
 
         with open(build_file_path, "w") as f:
-            to_write = json.loads(json.dumps(build_config.dict(), cls=EnumEncoder))
+            to_write = json.loads(
+                json.dumps(build_config.model_dump(), cls=EnumEncoder)
+            )
```

Comment:
actually a simpler version is `json.loads(build_config.json())` avoids the whole enum issue since Pydantic deals with it

Reply:
ah, shall we delete EnumEncoder all together?

Reply:
yeah we should! and `model.json()` is apparently deprecated as a method. we need to use `model.model_dump_json()`